blazar_dashboard/static/leases/js/lease_gantt.js

```diff
@@ -1,132 +1,221 @@
-      chooser.append(new Option('All Nodes', '*'));
-      nodeTypesPretty.forEach(function(nt) {
-        if (availableNodeTypes[nt[0]]) {
-          chooser.append(new Option(nt[1], nt[0]));
-          delete availableNodeTypes[nt[0]];
-        }
-      });
-      // fill chooser with node-types without a pretty name (when new ones pop up)
-      Object.keys(availableNodeTypes).forEach(function (key) {
-        if (availableNodeTypes[key]) {
-          chooser.append(new Option(key, key));
+        // populate resource-type-chooser
+        var chooser = $("#resource-type-chooser");
+        if(populateChooser != undefined){
+          $("label[for='resource-type-chooser']").text(chooser_attr_pretty);
+          var availableResourceTypes = {};
+          resp.resources.forEach(function(resource) {
+              availableResourceTypes[resource[chooser_attr]] = true;
+          });
+          chooser.empty();
+          chooser.append(new Option(`${gettext("All")} ${plural_resource_type}`, '*'));
+          populateChooser(chooser, availableResourceTypes)
+          Object.keys(availableResourceTypes).forEach(function (key) {
+            chooser.append(new Option(key, key));
+          });
+          chooser.prop('disabled', false);
+          chooser.change(function() {
+            var chosen_type = $('#resource-type-chooser').val();
+            filtered_reservations = all_reservations.map(function (reservation) {
+              var reservation_copy = Object.assign({}, reservation)
+              reservation_copy.data = reservation.data.filter(function(resource){
+                return chosen_type === '*' || chosen_type === resource[chooser_attr];
+              })
+              return reservation_copy
+            })
+            chart.updateOptions({series: filtered_reservations})
+            setTimeDomain(getTimeDomain())
+          });
+        } else {
+          chooser.hide()
         }
-      });
-      chooser.prop('disabled', false);
-
-      var taskNames = $.map(resp.compute_hosts, function(host, i) {
-        return host.node_name;
-      });
-      /* set initial time range */
-      var timeDomain = computeTimeDomain(7);
-
-      $('#blazar-gantt').empty().height(20 * taskNames.length);
-      gantt = d3.gantt({
-        selector: '#blazar-gantt',
-        taskTypes: taskNames,
-        taskStatus: taskStatus,
-        tickFormat: format,
-        timeDomainStart: timeDomain[0],
-        timeDomainEnd: timeDomain[1]
-      });
-      gantt(tasks);
-      setTimeDomain(timeDomain);
+        construct_calendar(filtered_reservations, computeTimeDomain(7))
     })
     .fail(function() {
-      $('#blazar-gantt').html('<div class="alert alert-danger">Unable to load reservations.</div>');
+      calendar_element.html(`<div class="alert alert-danger">${gettext("Unable to load reservations")}.</div>`);
     });
 
+    function construct_calendar(rows, timeDomain){
```

Comment:
nit: in Javascript functions and variables are typically camelCase, not snake_case, as you see elsewhere in the code.